optimism/src/keccak/witness.rs

```diff
@@ -10,18 +10,19 @@ use crate::{
     keccak::{
         column::KeccakWitness, interpreter::KeccakInterpreter, Constraint, Error, KeccakColumn,
     },
-    lookups::Lookup,
+    lookups::{FixedLookupTables, Lookup, LookupTable, LookupTableIDs::*},
 };
 use ark_ff::Field;
 use kimchi::o1_utils::Two;
+use kimchi_msm::LookupTableID;
 
 /// This struct contains all that needs to be kept track of during the execution of the Keccak step interpreter
 #[derive(Clone, Debug)]
 pub struct Env<Fp> {
     /// The full state of the Keccak gate (witness)
     pub witness: KeccakWitness<Fp>,
-    // The multiplicities of each lookup table
-    // TODO
+    /// The multiplicities of each lookup entry
+    pub multiplicities: Vec<Vec<u32>>,
```

Comment:
For further generalization: using a hashmap where the key is the ID of the table. From there, we can remove the assumptions made below that the table ID is the index in this vector.

Reply:
And, also from there, we can append any table, and make the code generic for other circuits.